compiler/src/dotty/tools/dotc/quoted/QuoteDriver.scala

```diff
@@ -62,8 +62,12 @@ class QuoteDriver extends Driver {
     output.getOrElse(throw new Exception("Could not extract " + expr))
   }
 
+  private[this] val myDefinitions = new ThreadLocal[Definitions] {
```

Comment:
Any reasons not to us a synchronised value shared between the threads?

Reply:
To be able to run several compiler instances in parallel. Different threads might try to initialize lazy val in Definitions at the same time.